File: classTypes4/java/src/Leon.java
// el leon hereda de CarnivoroTerrestre que a su vez hereda de Mamifero y AnimalBase
public class Leon extends CarnivoroTerrestre {

    // constructor con parametros
    public Leon(String nombreCientifico, double pesoKg, int gestacionMeses, String nombre) {
        // llama al constructor de la clase padre (CarnivoroTerrestre/Mamifero/AnimalBase)
        super(nombreCientifico, pesoKg, gestacionMeses);
    }

    // metodos obligatorios implementados
    @Override
    public void emitirSonido() {
        System.out.println("ROAR!!!");
    }

    @Override
    public void tecnicaDeCaza() {
        System.out.println("el leon caza en manada, acorralando a la presa");
    }
}

File: classTypes4/c#/Leon.cs
// el leon hereda de carnivoroterrestre que a su vez hereda de mamifero y animalbase
public class Leon : CarnivoroTerrestre{

    // constructor con parametros
    public Leon(string nombreCientifico, double pesoKg, int gestacionMeses, string nombre) 
        // llama al constructor de la clase padre
        : base(nombreCientifico, pesoKg, gestacionMeses) {
    }

    // metodos obligatorios implementados
    public override void emitirsonido(){
        Console.WriteLine("roar!!!");
    }

    public override void tecnicadecaza(){
        Console.WriteLine("el leon caza en manada, acorralando a la presa");
    }
}